spinn_utilities/ranged/ranged_list.py

```diff
@@ -240,7 +241,7 @@ def is_list(value, size):  # @UnusedVariable
         case as_list must also be extended
         """
         # Assume any iterable is a list
-        return hasattr(value, '__iter__')
+        return not is_singleton(value)
```

Comment:
OK, but it isn't quite the same.  Question is does it matter?

Reply:
Well, the new version is _particularly_ careful about not treating strings as lists (which was causing some strange test failures later), which they otherwise are in Python 3. The old test was adequate, but no longer is. Talk to @Christian-B about it for more details.

Reply:
OK, yes that is a good point!

Reply:
The ranged list needed to handle many different cases where the user do thing listL
pop_1.set(i_offset=new_i_offset)

new_i_offset could be (amongst others)
1. a single value 
     1a. While in pynn strings don't make sense I did not want to exclude them here
2. An iterator which provides 1 value for each neurons
3. a lambda function with the neuron id as the input (Pynn specification)
4. A pynn Random.

Note: 3 is handled by a PR merged in today
4 is Handled by a Extension class in Spynnaker

====
In the case of 1 all values are set to that value
In 2 we then do value[i] = list[i]
3: value[i] = function(i)
4. value[i] = rand.next()

if the value is a string I would not expect value[i] = the_string[i] but rather value[i] = the_string